src/frontend/screens/Audio/useAudioPlayback.ts

```diff
@@ -0,0 +1,85 @@
+import {Audio, AVPlaybackStatus, AVPlaybackStatusSuccess} from 'expo-av';
+import {useCallback, useEffect, useState, useRef} from 'react';
+import {Sound} from 'expo-av/build/Audio/Sound';
+
+export const useAudioPlayback = (recordingUri: string) => {
+  const recordedSoundRef = useRef<Sound | null>(null);
+  const [isPlaying, setPlaying] = useState(false);
+  const [duration, setDuration] = useState<number>(0);
+  const [currentPosition, setCurrentPosition] = useState(0);
+  const [isReady, setReady] = useState(false);
+
+  const audioCallbackHandler = useCallback((status: AVPlaybackStatus) => {
+    const update = status as AVPlaybackStatusSuccess;
+    if (update.didJustFinish) {
+      setPlaying(false);
+      setCurrentPosition(update.durationMillis ?? 0);
+    } else {
+      setPlaying(update.isPlaying);
+      if (update.isPlaying) {
+        setCurrentPosition(update.positionMillis);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    let soundInstance: Sound | null = null;
+    Audio.Sound.createAsync({uri: recordingUri})
+      .then(({sound, status}) => {
+        if ('error' in status && status.error) {
+          console.error('Error while creating audio playback', status.error);
+          return;
+        }
```

Comment:
as a follow up pr, i think we should handle audio errors (because we do not handle them in the UI at all)

Reply:
Ok